### ISSUE TITLE
Intel WiFi Not Working

### ISSUE THREAD

**Issue by Yardnuk** [gnodipac886/MatebookXPro-hackintosh#115]
I followed the instructions listed in README.md, and I still cannot get the WiFi to work. I keep getting the following message in the terminal: "mv: rename ./AppleIntelWiFi.kext to /temp/AppleIntelWiFi.kext: Directory not empty". I'm not sure what to do about this, so I thought I would ask. Thank you!

**Comment by CubeEON** [comment]
> I followed the instructions listed in README.md, and I still cannot get the WiFi to work. I keep getting the following message in the terminal: "mv: rename ./AppleIntelWiFi.kext to /temp/AppleIntelWiFi.kext: Directory not empty". I'm not sure what to do about this, so I thought I would ask. Thank you!

This is happening if you already did the process, the File is already existing in your /tmp folder so when running the script again it tells you that you alread have a file there, just reboot your system and the file will be gone because of the temporary characteristics of this directory.

**Comment by CubeEON** [comment]
But i too have problems with the WiFi, it just doenst seem to start up :/ already tried most of the things :/ 
Rebooted a dozen times, manually ran all the commands, no errors whatsoever but the wifi doesnt pop up in the Network settings, already tried profzei's wifi scripts but same thing, it just doesnt work. In the meantime i really think it got something to do with my Home Network?

**Comment by Yardnuk** [comment]
CubeEON, I'm having the EXACT same issue! I have tried four different methods trying to get the WiFi to work, and no matter what I do, I can never get the WiFi to show up in the Network settings! I'm completely lost...

**Comment by WhenKittensATK** [comment]
I'm having similar issues.
`user@Users-MBP Intel-WiFi % sh wifiLaunch.command
Type in WiFi name in exact format:
WiFi 5 GHz              
Type in WiFi password:
rm: AppleIntelWiFi.kext-0523.zip: No such file or directory
 
zip error: Nothing to do! (try: zip -r -q AppleIntelWiFi.kext-0523.zip . -i AppleIntelWiFi.kext)
 
mv: rename ./AppleIntelWiFi.kext to /tmp/AppleIntelWiFi.kext: No such file or directory
chown: /tmp/AppleIntelWiFi.kext: No such file or directory
/tmp/AppleIntelWiFi.kext failed to load - (libkern/kext) not found; check the system/kernel logs for errors or try kextutil(8).
user@Users-MBP Intel-WiFi % sh loadWiFi.sh
replace AppleIntelWiFi.kext/Contents/Info.plist? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
rm: AppleIntelWiFi.kext: No such file or directory`

**Comment by CubeEON** [comment]
> I'm having similar issues.
> 
> user@Users-MBP Intel-WiFi % sh wifiLaunch.command
> Type in WiFi name in exact format:
> WiFi 5 GHz
> Type in WiFi password:
> rm: AppleIntelWiFi.kext-0523.zip: No such file or directory
> 
> zip error: Nothing to do! (try: zip -r -q AppleIntelWiFi.kext-0523.zip . -i AppleIntelWiFi.kext)
> 
> mv: rename ./AppleIntelWiFi.kext to /tmp/AppleIntelWiFi.kext: No such file or directory
> chown: /tmp/AppleIntelWiFi.kext: No such file or directory
> /tmp/AppleIntelWiFi.kext failed to load - (libkern/kext) not found; check the system/kernel logs for errors or try kextutil(8).
> user@Users-MBP Intel-WiFi % sh loadWiFi.sh
> replace AppleIntelWiFi.kext/Contents/Info.plist? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
> rm: AppleIntelWiFi.kext: No such file or directory

Seems like in your case there is an error with copying the .kext to the /tmp folder. Do you have the .zip and the Scripts in the same folder, this should solve your problems :)

**Comment by WhenKittensATK** [comment]
Yes all the files are in the same folder.

**Comment by WhenKittensATK** [comment]
Yes I'm still having issues with the script. It's outputting errors like it doesn't see the files.

**Comment by EncryptedCurse** [comment]
Even without the script, I can't get it working at all. After manually editing the network name and password, I ran this:
```shell
sudo chmod -R 755 AppleIntelWiFi.kext
sudo chown -R root:wheel AppleIntelWiFi.kext
sudo kextload AppleIntelWiFi.kext
```
Nothing shows up in the menu bar icons or in System Preferences. Are there any other prerequisites (e.g. Clover config edits or kexts that I may have missed) that are needed for the WiFi to work?

**Comment by WhenKittensATK** [comment]
I got it to work following profzei's [guide](https://github.com/profzei/Matebook-X-Pro-2018/tree/master/Network) a few days ago.

**Comment by alex-spataru** [comment]
The kext provided by the repository did not work for me, and neither did the [latest version](https://github.com/AppleIntelWifi/adapter). In the latest version, the computer is able to scan for WiFi networks, however, I am unable to connect to any network. On the other hand, [itlwm](https://github.com/zxystd/itlwm) works perfectly for me, even if I load it directly from Clover. Also itlwm has more recent activity than [AppleIntelWifi](https://github.com/AppleIntelWifi/adapter). The only downside is that I must manually input the SSID/password in the kext.

**Comment by alex-spataru** [comment]
**Update**: you can use [HeliPort](https://github.com/zxystd/HeliPort) to manage the WiFi connections with [itlwm](https://github.com/zxystd/itlwm) without the need of modifying the kext.  However, the project is still in an early development stage. If you don't want to compile the project by yourself, a build from 23/Jun/2020 is available [here](https://files.gitter.im/5ecde0a9d73408ce4fe4fe2d/jH7z/HeliPort_23_06.zip).

**Comment by jmalfara** [comment]
Im facing the same issues here. zip error: Nothing to do! (try: zip -r -q AppleIntelWiFi.kext-0523.zip . -i AppleIntelWiFi.kext)

MBXP i7 16gb
